cif_pow.dic

```diff
@@ -3110,6 +3110,557 @@ save_pd_calib_wavelength.special_details
+          d   12.00   0.03   B
+          ...
+;
+;
+         Calibration offsets and the nominal position values to which they apply
+         are listed. The given _pd_calib_xcoord.overall_id refers to the
+         PD_CALIB_XCOORD_OVERALL category items given above.
+
+         An interpolation must be made to obtain a value for each data point.
+;
+
+save_
+
+save_pd_calib_xcoord.2theta_nominal
+
+    _definition.id                '_pd_calib_xcoord.2theta_nominal'
+    _definition.update            2023-01-17
+    _description.text
+;
+    The nominal 2\q value to which the offset given in
+    _pd_calib_xcoord.2theta_offset applies.
+;
+    _name.category_id             pd_calib_xcoord
+    _name.object_id               2theta_nominal
+    _type.purpose                 Number
+    _type.source                  Derived
+    _type.container               Single
+    _type.contents                Real
+    _enumeration.range            -180.0:180.0
+    _units.code                   degrees
+
+save_
+
+save_pd_calib_xcoord.2theta_offset
+
+    _definition.id                '_pd_calib_xcoord.2theta_offset'
+    _definition.update            2023-01-17
+    _description.text
+;
+    _pd_calib_xcoord.2theta_offset defines an offset angle (in degrees) used
+    to calibrate 2\q (as defined in _pd_meas.2theta_). Calibration is done
+    by adding the offset:
+
+         2\q~calibrated~ = 2\q~measured~ + 2\q~offset~
+;
+    _name.category_id             pd_calib_xcoord
+    _name.object_id               2theta_offset
+    _type.purpose                 Measurand
+    _type.source                  Derived
+    _type.container               Single
+    _type.contents                Real
+    _enumeration.range            -180.0:180.0
+    _units.code                   degrees
+
+save_
+
+save_pd_calib_xcoord.2theta_offset_su
+
+    _definition.id                '_pd_calib_xcoord.2theta_offset_su'
+    _definition.update            2023-01-17
+    _description.text
+;
+    Standard uncertainty of _pd_calib_xcoord.2theta_offset.
+;
+    _name.category_id             pd_calib_xcoord
+    _name.object_id               2theta_offset_su
+    _name.linked_item_id          '_pd_calib_xcoord.2theta_offset'
+    _units.code                   degrees
+
+    _import.get                   [{'file':templ_attr.cif  'save':general_su}]
+
+save_
+
+save_pd_calib_xcoord.id
+
+    _definition.id                '_pd_calib_xcoord.id'
+    _definition.update            2023-01-17
+    _description.text
+;
+    A code to uniquely identify each X-coordinate calibration.
+;
+    _name.category_id             pd_calib_xcoord
+    _name.object_id               id
+    _type.purpose                 Key
+    _type.source                  Assigned
+    _type.container               Single
+    _type.contents                Text
+    _method.purpose               Definition
+    _method.expression
+;
+    _enumeration.default = _pd_calib_xcoord.point_id
+;
```

Comment:
This is the only thing that worries me. Certainly the intention is that a missing `id` can be replaced by a `point_id`, which is great. However, what if some rows express the coordinate as a `point_id`, and others use straight `id`? In this case there is the danger that a `point_id` will be equal to an `id` used in a different row, violating the principle that each key value should be unique.

We could simply drop this dREL expression and require in all cases that an `id` is provided.

One fancy solution (that I've now worked out is probably wrong - see below) would make `pd_calib_xcoord` a child category of `pd_data`, and then `_pd_calib_xcoord.point_id` is both the key, and a child of `_pd_data.point_id` and so should not conflict. However, if we do this then we must necessarily have `_pd_calib_xcoord.diffractogram_id` as a co-key data name to match the key data names of `PD_DATA`. This is not a particularly onerous requirement, but it does imply that calibration is a per-diffractogram operation, which is not true, and so I've rejected it but leave the thinking here for posterity.

Reply:
Consider it dropped! :)